tests/expressions/tests.py

```diff
@@ -387,6 +387,125 @@ def test_ticket_18375_chained_filters(self):
+
+        error_message = 'This queryset contains a reference to an outer query, and may only be used in a subquery.'
+        with self.assertRaisesMessage(ValueError, error_message):
+            inner.exists()
+
+        self.assertTrue(outer.exists())
+
+    def test_subquery(self):
+        Company.objects.filter(name="Example Inc.").update(
+            point_of_contact=Employee.objects.get(firstname="Joe", lastname="Smith"),
+            ceo=Employee.objects.get(firstname="Max", lastname="Mustermann"),
+        )
+        Employee.objects.create(firstname="Bob", lastname="Brown", salary=40)
+        qs = Employee.objects.annotate(
+            is_point_of_contact=Exists(Company.objects.filter(point_of_contact=OuterRef('pk'))),
+            is_not_point_of_contact=~Exists(Company.objects.filter(point_of_contact=OuterRef('pk'))),
+            is_ceo_of_small_company=Exists(Company.objects.filter(num_employees__lt=200, ceo=OuterRef('pk'))),
+            is_ceo_small_2=~~Exists(Company.objects.filter(num_employees__lt=200, ceo=OuterRef('pk'))),
+            largest_company=SubQuery(Company.objects.order_by('-num_employees').filter(
+                models.Q(ceo=OuterRef('pk')) | models.Q(point_of_contact=OuterRef('pk'))
+            ).values('name')[:1], output_field=models.CharField())
+        ).values(
+            'firstname',
+            'is_point_of_contact',
+            'is_not_point_of_contact',
+            'is_ceo_of_small_company',
+            'is_ceo_small_2',
+            'largest_company'
+        ).order_by('firstname')
+
+        self.assertEqual([
+            {
+                'firstname': 'Bob',
+                'is_point_of_contact': False,
+                'is_not_point_of_contact': True,
+                'is_ceo_of_small_company': False,
+                'is_ceo_small_2': False,
+                'largest_company': None
+            },
+            {
+                'firstname': 'Frank',
+                'is_point_of_contact': False,
+                'is_not_point_of_contact': True,
+                'is_ceo_of_small_company': True,
+                'is_ceo_small_2': True,
+                'largest_company': 'Foobar Ltd.',
+            },
+            {
+                'firstname': 'Joe',
+                'is_point_of_contact': True,
+                'is_not_point_of_contact': False,
+                'is_ceo_of_small_company': False,
+                'is_ceo_small_2': False,
+                'largest_company': 'Example Inc.',
+            },
+            {
+                'firstname': 'Max',
+                'is_point_of_contact': False,
+                'is_not_point_of_contact': True,
+                'is_ceo_of_small_company': True,
+                'is_ceo_small_2': True,
+                'largest_company': 'Example Inc.'
+            }
+        ], list(qs))
+
+        self.assertEqual(1, qs.values('pk').filter(is_point_of_contact=True).count())
```

Comment:
The assertion argument order you've chosen is opposite of what we usally do, e.g.  `self.assertEqual(qs.values('pk').filter(is_point_of_contact=True).count(), 1)`